p2p/CHANGELOG.md

```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.0.14] - 2022-12-28
+
+- `ActorsToAddrBook` now skips actors that are not validators since they don't have a serviceUrl generic parameter
```

Comment:
Currently, we use the serviceUrl to connect to peers and to send messages to them, I believe that the `serviceUrl` is likely gonna be replaced by something like a LibP2P Address

https://docs.libp2p.io/concepts/fundamentals/addressing/

Eg: 
`/ip4/7.7.7.7/tcp/4242/p2p/QmRelay/p2p-circuit/p2p/QmRelayedPeer`

but that's going to be dynamic and transient and I don't see how it would be part of the state.

That said, I am not sure we should change the genesis state at all to handle addressing. 
It's the addressing that has to change instead.

Also, what is this concept of generic parameter? Where can I learn more about it?

Reply:
Regarding libp2p - makes sense when we get there.

Regarding the val filtering - see the comment I left in `p2p/addrbook_provider/persistence.go`

-----

`Also, what is this concept of generic parameter? Where can I learn more about it?`

This is just a poor design decision that was the result of a compromise during the shared module refactor.

We (currently) have 4 protocol actor types:
- Fisherman
- Validator
- Servicer
- Application

There are things they all share (e.g. an address) and there are things that are specific:
- Servicers need a "serviceURL" to service nodes
- Fisherman need a "serviceURL" to enable QoS
- Applications need a 'maxRelaysCount' which is only specific to them

The two potential future paths we take are:
1. One Protobuf per actor
2. One `BaseActor` protobuf that's embedded in each ProtocolActor type

See #313 for the origin document.

Reply:
Thanks for the wholesome explanation and context🙏

Reply:
🙏